.github/workflows/ci.yml

```diff
@@ -81,14 +81,40 @@ jobs:
           key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
           restore-keys: |
             ${{ runner.ox }}-yarn-
-      - name: Run build
-        run: yarn build
+      - name: Run build Shared components
+        run: yarn build:shared
+      - name: Run build Components
+        run: yarn build:components
       - name: Run tests
         run: yarn test
 
+  build_doc:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [ 16 ]
+    needs: [install-cache, build_and_test_components]
+    steps:
+      - name: Checkout Commit
+        uses: actions/checkout@v3
+      - name: Use Node.js ${{ matrix.node }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node }}
```

Comment:
Her you use `node`, but in the matrix you setup `node-version`. So even if you specify version `v16`, you will use `v18` later
![image](https://user-images.githubusercontent.com/4323800/221266820-4e80720a-17d7-45ae-9e5f-9ecf86959655.png)